src/CRC.cpp

```diff
@@ -37,8 +41,24 @@ u32 CRC_Calculate( u32 crc, const void * buffer, u32 count )
 	u32 orig = crc;
 
 	p = (u8*) buffer;
+
+#if defined(__ARM_FEATURE_CRC32)
+	// use four byte crc intrinsic __crc32w if count is high enough.
+	while (count >= 4) {
+		crc = __crc32w(crc, (p[0]  | p[1] << 8 | p[2] << 16 | p[3] << 24));
```

Comment:
Thank you very much. I have tried to use this but had done something wrong. I just looked how the linux kernel is doing it. It just shifts bytes. Your solution runs much better. I can see a speed gain of factor 9.3.